doc/nyan/base types.md

```diff
@@ -0,0 +1,164 @@
+
+	allowedResource : set(Resource) # Allowed ressources?
+	maxAmount : int # maximum amount  that can be carried
+	speed : float # gathered units per second
+
+	# WIP2: proposal: allow different maximum amounts for different resources:
+
+#	allowedResource : orderedset(Resource)
+#	maxAmount : orderedSet(int) # does <10,5,0,10> work? it is a set after all.
+
+	# WIP3: if WIP2 is not possible: implement an 'Ability' that allows to gather only one resource.
+	# A worker would then need 4 gather abilities added to it's 'abilities' set.
+	# And this would allow to reference
+
+	# WIP4: how do we reference Animations?
+#	animation : file # one picture that contains all frames or, if a directory,
+	                # contains all frames. # WIP5: naming scheme?
+#	renameMe : file # WIP6: how to read `animation'? describes, where in the big picture `animation' which frames are; whether to potentially mirror them.
+	                # possibilities: 1. $renameMe is a regular file: description of `animation' (syntax?)
+	                # 2. replace `renameMe' by more fields that describe `animation' (ugly imo, but maybe the best way)
+	                # 3. remove `renameMe':
+	                #    if $animation is a directory: look for '$animation/animate.nfo'
+	                #    else: look for "$(basename $animation png)nfo"
+
+CarryAbility(Ability): # WIP: find a better name?
+	# Used by priests for relicts.
+	# WIP1: how should this look like? keep it general? can maybe be used by buildings to let units find shelter in them? or by mods for adding new relict-like objects
+	# WIP2: can maybe merged with a modified GatherAbility?
+
+#	goodMemberName : Portable # what can be collected? make a 'Portable' class and let relict inherit that? could then be also used by resources?
+#	maxAmount
+
+	animation : file # what does a priest look like when he's carrying it?
+	                 # WIP: see GatherAbility.animation and nyan question 4 (further down).
+
+AttackAbility(Ability):
+	# Every attack shall inherit from this.
+	# WIP: how will damage calculation work in oa?
+
+	range : float
+	rate : float
+	damage : float # rename to baseDamage ?
+
+
+Unit(): # WIP
+	# Everything you can see in the game # (WIP: but not assets, which is basically everything you see. Maybe better:)
+	# Everything that has some 'Ability' inherits from this.
+
+	ability : set(Ability)
+
+
+
+#and probably more. if you know what's missing, add it.
+
+```
+
+Open nyan implementation questions:
+
+1. How do we say in Nyan, that a Towncenter attack depends on how many people are inside the building?
+2. What about the building 'Ability' to carry units? will this lead to recursive references?
+3. How do we reference hooks from Nyan/create new hooks in mods?
+
+  -> If the {Towncenter,Castle} has certain damage, contained units need to leave.
+  
+  -> A relict in a Church leads to increasing gold.
```

Comment:
The damage info is a combination of nyan and the engine. The building itself must define a collapse threshold (in percent maybe) which the engine respects to send units out.

The relic gold is purely implemented in the engine which provides a "ResourceGenerator" object, which the relic inherits (`Relic(ResourceGenerator)`) and adds to the `set(resource_type) += {Gold}` and `rate = 10`. Nyan is not aware of the game mechanics, this is always the job of the engine (that was by the way the key enlightment i finally had to get a sufficient nyan design).